pkg/build/kube/tarbits.go

```diff
@@ -0,0 +1,425 @@
+	RegisterNamedBits("tar", NewTarBits)
+}
+
+// NewTarBits returns a new Bits backed by a version file, the release image
+// archives and the kubeadm, kubectl, and kubelet binaries. The given tarRoot
+// may be defined three ways:
+//
+//   1. A local filesystem path containing the files listed above in a flat
+//      structure.
+//
+//   2. An HTTP address that contains the files listed above in a structure
+//      that adheres to the layout of the public GCS buckets kubernetes-release
+//      and kubernetes-release-dev. For example, if tarRoot is set to
+//      https://k8s.ci/v1.13 then the following URIs should be valid:
+//
+//        * https://k8s.ci/v1.13/kubernetes.tar.gz
+//        * https://k8s.ci/v1.13/bin/GOOS/GOARCH/kube-apiserver.tar
+//        * https://k8s.ci/v1.13/bin/GOOS/GOARCH/kube-controller-manager.tar
+//        * https://k8s.ci/v1.13/bin/GOOS/GOARCH/kube-scheduler.tar
+//        * https://k8s.ci/v1.13/bin/GOOS/GOARCH/kube-proxy.tar
+//        * https://k8s.ci/v1.13/bin/GOOS/GOARCH/kubeadm
+//        * https://k8s.ci/v1.13/bin/GOOS/GOARCH/kubectl
+//        * https://k8s.ci/v1.13/bin/GOOS/GOARCH/kubelet
+//
+//   3. A valid semantic version for a released version of Kubernetes or
+//      begins with "ci/" or "release/". If the string matches any of these
+//      then the value is presumed to be a CI or release build hosted on one
+//      of the public GCS buckets.
+//
+//      This option also supports values ended with ".txt", ex. "ci/latest.txt".
+//      In fact, all values beginning with "ci/" or "release/" are first checked
+//      to see if there's a matching txt file for that value. For example,
+//      if "ci/latest" is provided then before assuming that is a directory,
+//      "ci/latest.txt" is queried.
+//
+// In order to prevent the second two options from conflicting with the first,
+// a local file path, using the prefix "file://" explicitly indicates the given
+// tarRoot is a local file path.
+//
+// The given tarRoot may also take the format tarRoot::localPath where the
+// tarRoot is separated from the local path where the remote files will be
+// downloaded.
+//
+// If no explicit local path is provided then remote files are downloaded to
+// "$HOME/.kind/tarbits/VERSION/GOOS/GOARCH" where VERSION is the value
+// extracted from the "version" file in the "kubernetes.tar.gz" tarball.
+//
+// If a file is already present on disk with the same size as the remote
+// equivalent, the file will not be downloaded again.
+func NewTarBits(tarRoot string) (bits Bits, err error) {
+	tarBits := &TarBits{}
+
+	if strings.HasPrefix(tarRoot, "file://") {
+		tarBits.tarRoot = strings.Replace(tarRoot, "file://", "", 1)
+		tarBits.tbitsType = tarBitsLocal
+	} else if strings.HasPrefix(tarRoot, "http:") ||
+		strings.HasPrefix(tarRoot, "https:") {
+		tarBits.remoteURI = tarRoot
+		tarBits.tbitsType = tarBitsHTTP
+	} else if ok, _ := regexp.MatchString(
+		`^(v\d+\.\d+\.\d+[\w.\-+]*)(::.+)?$`, tarRoot); ok {
```

Comment:
better to use the functions from here:
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/version/version.go

the package is already vendored in kind.